### ISSUE TITLE
No graphic for last version?

### ISSUE THREAD

**Issue by melsophos** [bcpierce00/unison#213]
I am trying to compile the unison with the graphic ui (version 2.51.2) on Fedora 28. For some context, the only available version through DNF is 2.40.128 and it is buggy (lost connection with server error). I was relying on older packages (for Fedora 20) but they disappeared recently.

It does not seem to me that the graphic version is generated. Indeed, when i do `make` or `make UISTYLE=gtk2` I get a binary which is text only, despite the fact that I installed all packages related to ocaml and gtk (`ocaml-lablgtk ocaml-lablgtk-devel gtk2-devel`). Am I missing something?

**Comment by cookiemonsteruk** [comment]
I've been struggling with this myself. Finally solved with by looking carefully at the make output. I had the error ` 473: File "/home/myuser/Downloads/unison-master/src/uigtk2.ml", line 97, characters 5-29:
  474  Error: Unbound value Gpointer.region_of_bytes
  475: Makefile.OCaml:423: recipe for target 'uigtk2.cmx' failed
  476: make[1]: *** [uigtk2.cmx] Error 2
  477  make[1]: Leaving directory '/home/myuser/Downloads/unison-master/src'
  478  Makefile:14: recipe for target 'src' failed`
An internet search led me to the post by bcpierce here "https://www.onwebsecurity.com/configuration/its-about-the-journey-compiling-64-bit-unison-gtk2-on-windows.html". I'm so glad he documented it.
Took me to the commit and what I did was change in the file on the line 473 uigtk2.ml from region_of_bytes to region_of_string. Save and run "make" again. This time built finally for gtk2.
Thanks Benjamin for documenting it!
My system is Ubuntu 16.04 but might work for you too mesophos. I also had to install the various packages you did. I also had to "force" with: make UISTYLE=gtk2.
Downloaded unison from master branch as of 30/09/2018

**Comment by PeterMosmans** [comment]
@cookiemonsteruk Although the post "https://www.onwebsecurity.com/configuration/its-about-the-journey-compiling-64-bit-unison-gtk2-on-windows.html" is not by @bcpierce00  - thanks for the kind words :wink:
And of course thanks @bcpierce00 for such an amazing piece of software.

**Comment by melsophos** [comment]
So it took me some time, but thanks to the comment from @cookiemonsteruk and the post from @PeterMosmans I managed to compile the version 2.48.15 (from the [tar.gz source here](https://github.com/bcpierce00/unison/releases/tag/v2.48.15v4)).

One weird thing is that I had to replace `region_of_string` by `region_of_bytes` in Fedora 30, but not in Fedora 27 (the compilation went straightforwardly on the latter).

However, I still think that there is a problem with the makefile of the last version 2.51.2 when comparing with the makefile of 2.48.15. In particular, the default installation has `UISTYLE=text`, contrary to the [documentation](https://www.cis.upenn.edu/~bcpierce/unison/download/releases/stable/unison-manual.html#build-unix), which makes me think that it cannot build the graphic version.
